src/scheduler/simulate.c

```diff
@@ -1547,54 +1538,34 @@ add_timed_event(timed_event *events, timed_event *te)
  *
  * @param[in] sinfo    - sinfo which contains calendar to delete from
  * @param[in] e        - event to delete
- * @param[in] flags    - flag bitfield
- *						 DE_UNLINK - unlink event and don't free
  *
- * @return int
- * @retval 1 event was successfully deleted
- * @retval 0 failure
+ * @return void
  */
 
-int
-delete_event(server_info *sinfo, timed_event *e, unsigned int flags)
+void
+delete_event(server_info *sinfo, timed_event *e)
 {
-	timed_event *cur_e;
-	timed_event *prev_e = NULL;
 	event_list *calendar;
 
 	if (sinfo == NULL || e == NULL)
-		return 0;
+		return;
 
 	calendar = sinfo->calendar;
 
+	if (calendar->next_event == e)
+		calendar->next_event = e->next;
```

Comment:
What will happen to first_run_event if the event being deleted is the 'first_run_event'. I think we need to update that too.